tests/test_soap.py

```diff
@@ -60,7 +60,9 @@ def test_soap_can_mask_spatial_and_non_spatial_actually_use(filename):
     masses2 = data2.spherical_overdensity_200_mean.total_mass
 
     # Manually mask
-    custom_mask = (masses2 >= lower) & (masses2 <= upper)
+    custom_mask = (
+        unyt.unyt_array(masses2.to_value(masses2.units), masses2.units) >= lower
```

Comment:
Why is this needed, does masses2 not already support comparison?

Reply:
`lower` and `upper` are `unyt_quantity`s, not `cosmo_array`s. This is correct given the type hints of `constrain_mask` used further up in the test, but correctly triggers a warning on comparison with `masses2`. Perhaps `constrain_mask` should be modified to expect `cosmo_array`s. It would still work in that case but would raise warnings on receiving `unyt_quantity`s (which is what should happen in that case, I guess). If that happens then the change to `custom_mask` should be reverted (it will emit a warning in that case so shouldn't go unnoticed).